### DISCUSSION TITLE
How do I continue training SiamRPN after epoch 10?

### DISCUSSION THREAD

**Discussion by JayJayBinks1** [STVIR/pysot#428] [General] [answered]
What do I need to change in the config? Currently I have
PRETRAINED: pretrained_models/resnet50.model
RESUME: 'snapshot/checkpoint_10.pth'

However I get "ValueError: loaded state dict contains a parameter group that doesn't match the size of optimizer's group".

I've checked other threads with this issue but nothing works.

Did I set up the config correctly? Do I need to resume from checkpoint 10 or should that be the PRETRAINED setting?

**Comment by JayJayBinks1** [comment]
Also I have TRAIN.START_EPOCH: 10

**Reply by ssbagalkar** [reply]
Are you trying to `resume` training?

**Comment by JayJayBinks1** [comment]
Also, is it normal for the model to stop training once it finishes epoch 10?

**Reply by ssbagalkar** [reply]
You can change that in the `config.yaml ` under `TRAIN` set `EPOCH` to whatever you want as the end epoch number

**Comment by lb1100** [comment]
Actually, two stage do not share optimizers(stage 2 adds more trainable parameters). So the optimizer's group doesn't match. You can set TRAIN.START_EPOCH: 10 to resume training. 
We make two stage into single process to make scripts cleaner. In this way we do not need to start training process for two times. It will save a lot of time since starting dataloader will comsume a lot of time.

**Comment by lb1100** [comment, marked as answer]
duplicate #80

**Comment by JayJayBinks1** [comment]
I did see #80 but I'm wondering if it's normal for the script to stop after the 10 epoch?

**Reply by ZhiyuanChen** [reply]
You mean resuming?
